mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/HadoopFileSystemBlobStore.java

```diff
@@ -30,22 +33,32 @@
  * such as s3, gfs, etc...
  */
 @RequiredArgsConstructor
+@Slf4j
 public class HadoopFileSystemBlobStore implements BlobStore {
 
     //  The file system in which blobs are stored. */
     private final FileSystem fileSystem;
 
     private final File localStoreDir;
 
+    final Striped<Lock> locks = Striped.lock(10);
+
     @Override
     public File get(URI blobUrl) throws IOException {
         final Path src = new Path(blobUrl);
         final Path dest = new Path(getStorageLocation(blobUrl));
-        File destFile = new File(dest.toUri().getPath());
-        if (!destFile.exists()) {
-            fileSystem.copyToLocalFile(src, dest);
+        Lock lock = locks.get(dest);
+        try {
+            lock.lock();
+            log.info("Getting file with path {} from HadoopFileSystemBlobStore", dest);
```

Comment:
nit: HadoopFileSystemBlobStore should already be part of the logger classname. No need to add it here too.